### PR TITLE
MMV2-sdk-funs

### PR DESCRIPTION
**Pull request by pallavi-pannu-oc**
Added deployment api's

### CODE REVIEW

File: dkube/sdk/api.py [hunk 1]
```diff
@@ -2952,3 +2952,252 @@ def update_cluster_configuration(self, clustername, data):
...
+        if tags:
+            query_params["tags"] = tags
+        if page:
+            query_params["page"] = page
+        if archived:
+            query_params["archived"] = archived
+        response = self._api.list_deployments().to_dict()
+        return response["data"]
+
+    def get_deployment_id(self, name=None):
```
Comment (songole) [inline dkube/sdk/api.py:2980]: it should also take cluster name, namespace and variant  All of them could be null. 
if cluster name is null, namespace & variant are also null
if cluster name is specified, look up the cluster name. If it is AWS, variant should not be null. If it is remote-dkube, namespace should not be null

then, you can compare in the list.

File: dkube/sdk/api.py [hunk 2]
```diff
@@ -2952,3 +2952,252 @@ def update_cluster_configuration(self, clustername, data):
...
+        cluster=None,
+        deployment_url=None,
+        model_reference=None,
+        namespace=None,
+        variant=None,
+    ):
+        """
+         Method to import the deployment
+
+        *Available in DKube Release: 3.0*
```
Comment (songole) [inline dkube/sdk/api.py:3027]: available in 3.3. Please check the availability string in all APIs.

File: dkube/sdk/api.py [hunk 3]
```diff
@@ -2952,3 +2952,252 @@ def update_cluster_configuration(self, clustername, data):
...
+
+        Outputs*
+            a dictionary object with response status
+        """
+        response = self._api.archive_deployments(
+            data={"deployment_ids": [id]}, archive="false"
+        )
+        return response
+
+    def unarchive_deployments(self, ids=[]):
```
Comment (songole) [inline dkube/sdk/api.py:3187]: Do we have such APIs for jobs as well like one singular and another plural?

**Comment by pallavi-pannu-oc** [comment]
@songole suggested changes are done in this commit.

File: dkube/sdk/api.py [hunk 4]
```diff
@@ -2977,28 +2977,56 @@ def list_deployments(self, **kwargs):
...
+            for deployment in self.list_deployments():
+                if deployment["name"] == name:
+                    return deployment["id"]
+        else:
+            if clustername:
+                cluster_kind = self.get_cluster_details(clustername)["data"]["cluster"][
+                    "kind"
+                ]
+                if cluster_kind == "sagemaker" and variant == None:
+                    print("Please provide the variant for the AWS cluster first")
```
Comment (songole) [inline dkube/sdk/api.py:3012]: Can you remove 'first' at the end?

**Pull request merged**